ballerina/tests/test.bal

```diff
@@ -0,0 +1,224 @@
+
+// Record type to hold test data.
+public type TestData record {
+    string fileId;
+    string modelId;
+    string jobId;
+};
+
+// Initialize test data.
+TestData testData = {fileId: "", modelId: "", jobId: ""};
+
+// Function to generate test data.
+function dataGen() returns TestData[][] {
+    return [[testData]];
+}
+
+@test:Config {
+    dataProvider:  dataGen,
+    groups: ["Models"]
+}
+isolated function testListModels(TestData testData) returns error? {
+    ListModelsResponse modelsResponse = check openaiFinetunes->/models.get();
+    testData.modelId = "gpt-3.5-turbo";
+    test:assertEquals(modelsResponse.'object, "list", "Object type mismatched");
+    test:assertTrue(modelsResponse.hasKey("data"), "Response does not have the key 'data'");
+}
+
+@test:Config {
+    dataProvider:  dataGen,
+    dependsOn: [testListModels],
+    groups: ["Models"]
+}
+isolated function testRetrieveModel(TestData testData) returns error? {
+    string modelId = testData.modelId;
+    Model modelResponse = check openaiFinetunes->/models/[modelId].get();
+    test:assertEquals(modelResponse.id, modelId, "Model id mismatched");
+    test:assertTrue(modelResponse.hasKey("object"), "Response does not have the key 'object'");
+}
+
+@test:Config {
+    dependsOn: [testCreateFineTuningJob, testListModels, testRetrieveModel, testListFineTuningJobCheckpoints, testListFineTuningEvents],
+    dataProvider:  dataGen,
+    enable: isLiveServer? false : true, // Enable this test only for mock server.
+    groups: ["Models"]
+}
+isolated function testDeleteModel(TestData testData) returns error? {
+    string modelIdCreated = testData.modelId;
+    DeleteModelResponse modelResponseDelete = check openaiFinetunes->/models/[modelIdCreated].delete();
+    test:assertEquals(modelResponseDelete.id, modelIdCreated, "Model id mismatched");
+    test:assertTrue(modelResponseDelete.hasKey("object"), "Response does not have the key 'object'");
+}
+
+@test:Config {
+    groups: ["Files"]
+}
+isolated function testListFiles() returns error? {
+    ListFilesResponse filesResponse = check openaiFinetunes->/files.get();
+    test:assertEquals(filesResponse.'object, "list", "Object type mismatched");
+    test:assertTrue(filesResponse.hasKey("data"), "Response does not have the key 'data'");
+}
+
+@test:Config {
+    dependsOn: [testListFiles],
+    dataProvider: dataGen,
+    groups: ["Files"]
+}
+isolated function testCreateFile(TestData testData) returns error? {
+    CreateFileRequest fileRequest = {
+        file: {fileContent, fileName},
+        purpose: "fine-tune"
+    };
+
+    OpenAIFile fileResponse = check openaiFinetunes->/files.post(fileRequest);
+    testData.fileId = fileResponse.id;
+    test:assertEquals(fileResponse.purpose, "fine-tune", "Purpose mismatched");
+    test:assertTrue(fileResponse.id !is "", "File id is empty");
+}
+
+@test:Config {
+    dependsOn: [testCreateFile],
+    dataProvider:  dataGen,
```

Comment:
Incorrect formatting. Please format all the source files using Ballerina VSCode formatter
```suggestion
    dataProvider: dataGen,
```